src/youtube.js

```diff
@@ -0,0 +1,1092 @@
+   * @public
+   */
+  isAdaptiveBitrateEnabled(): boolean {
+    return this._isAdaptiveBitrate;
+  }
+
+  /**
+   * Seeking to live edge.
+   * @function seekToLiveEdge
+   * @returns {void}
+   * @public
+   */
+  seekToLiveEdge(): void {
+    if (this._api) {
+      this._api.seekTo(this._api.getDuration()); //TODO: need to check this
+    }
+  }
+
+  /**
+   * Get the start time of DVR window in live playback in seconds.
+   * @returns {Number} - start time of DVR window.
+   * @public
+   */
+  getStartTimeOfDvrWindow(): number {
+    return this._api ? this._api.getDuration() : 0; //TODO: need to fix this
+  }
+
+  /**
+   * Checking if the current playback is live.
+   * @function isLive
+   * @returns {boolean} - Whether playback is live.
+   * @public
+   */
+  isLive(): boolean {
+    return false; //TODO: how to know this? https://developers.google.com/youtube/2.0/developers_guide_protocol_retrieving_live_events?
+  }
+
+  /**
+   * Start/resume playback.
+   * @public
+   * @returns {void}
+   */
+  play(): void {
+    if (this._api) {
+      if (this.currentTime === this.duration) {
+        this.currentTime = 0;
+      }
+      this._api.playVideo();
+      // this.dispatchEvent(new FakeEvent(EventType.PLAY));
+    }
+  }
+
+  /**
+   * Pause playback.
+   * @public
+   * @returns {void}
+   */
+  pause(): void {
+    if (this._api) {
+      this._api.pauseVideo();
+    }
+  }
+
+  /**
+   * Load media.
+   * @param {number} startTime - Optional time to start the video from.
+   * @public
+   * @returns {Promise<Object>} - The loaded data
+   */
+  load(startTime: ?number): Promise<Object> {
+    this._loaded = true;
+    this._sdkLoaded
+      .then(() => {
+        //TODO: if autoplay pass it here and check in play if already playing?
+        const loadOptions = {
+          videoId: this._source.url,
+          startSeconds: (startTime && startTime > 0) ? startTime : 0
+        };
+        this._api.loadVideoById(loadOptions);
+      })
+      .catch(error => {
+        return Promise.reject(error);
+      });
+    this._videoLoaded = {};
+    this._videoLoaded.promise = new Promise((resolve, reject) => {
+      this._videoLoaded.resolve = resolve;
+      this._videoLoaded.reject = reject;
+    });
+    return this._videoLoaded.promise;
+  }
+
+  /**
+   * Parse hls video tracks into player video tracks.
+   * @returns {Array<VideoTrack>} - The parsed video tracks.
+   * @private
+   */
+  _parseVideoTracks(): Array<VideoTrack>{
+    const levels: Array<string> = this._api.getAvailableQualityLevels();
+    const currentLevel: string = this._api.getPlaybackQuality();
+    const videoTracks = [];
+    levels.forEach((level, i) => {
+      let settings = {
+        active: currentLevel === level,
+        label: level,
+        bandwidth: 0,
+        width: i+1, //TODO: get estimated width and height from https://developers.google.com/youtube/iframe_api_reference#Playback_quality
```

Comment:
i don't understand this, i is the index in the getAvailableQualityLevels array, right?

Reply:
For now this method is a placeholder as it seems the `setQualityLevel` method doesn't even work.
If we will have it working then we can take the suggested height/width values from the linked URL, where youtube has some outline of quality name to width/height values.